File: solutions/infy/timed_product.php
<?php
/* version 1: without using any php extensions. */
/* WARNING: Make sure to comment out one of these to variants before executing. */
preg_match('/([0-9]{1,4}) x ([0-9]{1,4})/', file_get_contents('http://directus.darkscience.net:6789/'), $res);
if(is_array($res)) { $product = $res[1]*$res[2]; }

$c = http_build_query(array('answer' => $product));

$s = fsockopen('directus.darkscience.net', 6789);
fwrite($s, "POST / HTTP/1.1\r\n");
fwrite($s, "Host: directus.darkscience.net\r\n");
fwrite($s, "Content-Type: application/x-www-form-urlencoded\r\n");
fwrite($s, "Content-Length: " . strlen($c) . "\r\n");
fwrite($s, "Connection: close\r\n");
fwrite($s, "\r\n");
fwrite($s, $c);

$o = '';
while(!feof($s)) { $o .= fgets($s, 1024); }
fclose($s);

if(preg_match('/Correct/', $o)) { echo 'Correct'; }else { echo 'Wrong'; }





/* version 2: using the cURL extension */
preg_match('/([0-9]{1,4}) x ([0-9]{1,4})/', file_get_contents('http://directus.darkscience.net:6789/'), $res);
if(is_array($res)) { $product = $res[1]*$res[2]; }

$c = http_build_query(array('answer' => $product));

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, 'http://directus.darkscience.net:6789/');
curl_setopt($ch, CURLOPT_HTTPHEADER, array(
    'Host: directus.darkscience.net',
    'Content-Type: application/x-www-form-urlencoded',
    'Content-Length: ' . strlen($c),
    'Connection: close'
));
curl_setopt($ch, CURLOPT_POST, true);
curl_setopt($ch, CURLOPT_POSTFIELDS, $c);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

$out = curl_exec($ch);
curl_close($ch);

if(strstr($out, 'Correct')) { echo 'Correct!'; }else { echo 'Wrong.'; }

File: solutions/infy/timed_product.py
#!/usr/bin/python

import httplib
import urllib2
import re

m = re.findall('([0-9]{1,4}) x ([0-9]{1,4})', urllib2.urlopen('http://directus.darkscience.net:6789/').read())
product = str(int(m[0][0])*int(m[0][1]))

s = 'answer=' + product

headers = {
    "Host": "directus.darkscience.net",
    "Content-Type": "application/x-www-form-urlencoded",
    "Content-Length": len(s),
    "Connection": "close"
}

c = httplib.HTTPConnection("directus.darkscience.net:6789")
c.request('POST', "/", s, headers)
response = c.getresponse()
data = response.read()

if data.find('Correct'):
    print 'Correct!'
else:
    print 'Wrong.'

c.close()